ethereum/forge-test/Bridge.t.sol

```diff
@@ -56,6 +146,1289 @@ contract TestBridge is Test {
+                0x54657374546f6b656e0000000000000000000000000000000000000000000000
+            )
+        );
+        vm.expectEmit();
+        emit LogMessagePublished(
+            address(bridge),
+            uint64(0),
+            uint32(234),
+            attestPayload,
+            uint8(finality)
+        );
+        bridge.attestToken(address(tokenImpl), 234);
+    }
+
+    function testShouldCorrectlyDeployAWrappedAssetForATokenAttestation()
+        public
+    {
+        testShouldRegisterAForeignBridgeImplementationCorrectly();
+        testShouldAttestATokenCorrectly();
+        bytes memory data = abi.encodePacked(
+            uint8(2),
+            testBridgedAssetAddress,
+            testBridgedAssetChain,
+            uint8(18),
+            // symbol (TT)
+            bytes32(
+                0x5454000000000000000000000000000000000000000000000000000000000000
+            ),
+            // name (TestToken)
+            bytes32(
+                0x54657374546f6b656e0000000000000000000000000000000000000000000000
+            )
+        );
+        bytes memory vaa = signAndEncodeVM(
+            0,
+            0,
+            testForeignChainId,
+            testForeignBridgeContract,
+            0,
+            data,
+            uint256Array(testGuardian),
+            0,
+            0
+        );
+
+        bridge.createWrapped(vaa);
+        address wrappedAddress = bridge.wrappedAsset(
+            testBridgedAssetChain,
+            testBridgedAssetAddress
+        );
+        assertTrue(
+            bridge.isWrappedAsset(wrappedAddress),
+            "Wrapped asset is not a wrapped asset"
+        );
+
+        TokenImplementation wrapped = TokenImplementation(wrappedAddress);
+
+        assertEq(wrapped.symbol(), "TT");
+        assertEq(wrapped.name(), "TestToken");
+        assertEq(wrapped.decimals(), 18);
+        assertEq(wrapped.chainId(), testBridgedAssetChain);
+        assertEq(wrapped.nativeContract(), testBridgedAssetAddress);
+    }
+
+    function testShouldCorrectlyUpdateAWrappedAssetForATokenAttestation()
+        public
+    {
+        testShouldCorrectlyDeployAWrappedAssetForATokenAttestation();
+        bytes memory data = abi.encodePacked(
+            uint8(2),
+            testBridgedAssetAddress,
+            testBridgedAssetChain,
+            uint8(18),
+            // symbol
+            bytes32(
+                0x5555000000000000000000000000000000000000000000000000000000000000
+            ),
+            // name
+            bytes32(
+                0x5472656500000000000000000000000000000000000000000000000000000000
+            )
+        );
+
+        bytes memory vaa = signAndEncodeVM(
+            0,
+            0,
+            testForeignChainId,
+            testForeignBridgeContract,
+            0,
+            data,
+            uint256Array(testGuardian),
+            0,
+            0
+        );
+
+        vm.expectRevert("current metadata is up to date");
```

Comment:
Separate out into a separate test.

Reply:
Agree this should be addressed with #4090